go/firewallsso/lib/base.go

```diff
@@ -0,0 +1,135 @@
+package libfirewallsso
+
+import (
+	"context"
+	"fmt"
+	log "github.com/inconshreveable/log15"
+	"github.com/inverse-inc/packetfence/go/logging"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+	"net"
+)
+
+// Basic interface that all FirewallSSO must implement
+type FirewallSSOInt interface {
+	init(ctx context.Context) error
+	logger(ctx context.Context) log.Logger
+	Start(ctx context.Context, info map[string]string, timeout int) bool
+	Stop(ctx context.Context, info map[string]string) bool
+	GetFirewallSSO(ctx context.Context) *FirewallSSO
+	IsRoleBased(ctx context.Context) bool
+	MatchesRole(ctx context.Context, info map[string]string) bool
+	MatchesNetwork(ctx context.Context, info map[string]string) bool
+}
+
+// Basic struct for all firewalls
+type FirewallSSO struct {
+	PfconfigMethod string `val:"hash_element"`
```

Comment:
I really doubt another library would lookup things in that field... but I can adjust it